File: types/history.ts
export interface PredictionHistory {
  id: string;
  userId: string;
  timestamp: string;
  createdAt: string;
  riskLevel: string;
  confidence: number;
  predictionSummary: string;
  imagePath?: string;
  processingTime: number;
  modelVersion: string;
  explanation?: string;
  dietRecommendation?: string;
}

export interface HistoryStatistics {
  totalScans: number;
  highestConfidence: number;
  averageConfidence: number;
  lowRiskPercent: number;
  moderateRiskPercent: number;
  highRiskPercent: number;
  latestScan?: string;
  firstScan?: string;
  mostCommonRisk?: string;
}

export interface TimelineEntry {
  predictionId: string;
  timestamp: string;
  riskLevel: string;
  confidence: number;
  modelVersion: string;
}

export interface HistoryResponse {
  success: boolean;
  predictions: PredictionHistory[];
  statistics: HistoryStatistics;
  timeline: TimelineEntry[];
  error?: string;
}

File: backend/app/schemas/history.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime

class PredictionHistoryCreate(BaseModel):
    userId: str
    riskLevel: str
    confidence: float
    predictionSummary: str = ""
    imagePath: Optional[str] = None
    processingTime: float
    modelVersion: str
    explanation: Optional[str] = None
    dietRecommendation: Optional[str] = None

class PredictionHistory(PredictionHistoryCreate):
    id: str
    timestamp: datetime
    createdAt: datetime

class HistoryStatistics(BaseModel):
    totalScans: int
    highestConfidence: float
    averageConfidence: float
    lowRiskPercent: float
    moderateRiskPercent: float
    highRiskPercent: float
    latestScan: Optional[datetime] = None
    firstScan: Optional[datetime] = None
    mostCommonRisk: Optional[str] = None

class TimelineEntry(BaseModel):
    predictionId: str
    timestamp: datetime
    riskLevel: str
    confidence: float
    modelVersion: str

class HistoryResponse(BaseModel):
    success: bool
    predictions: List[PredictionHistory]
    statistics: HistoryStatistics
    timeline: List[TimelineEntry]
    error: Optional[str] = None
